### PR TITLE
on join create guild config

### LINKED ISSUE #51: On bot join, initialize configs
In general: make sure the bot works from a clean setup.

### PR DESCRIPTION
**Pull request by PandaGamersHDNL**

### CODE REVIEW

**Comment by PandaGamersHDNL** [comment]
ready for review, open to changes :D

File: src/services/guild-service.ts [hunk 1]
```diff
@@ -25,9 +25,26 @@ export class GuildService {
...
 
-  static async updateCourseChannels(guildID: string, courseChannels: CourseChannels): Promise<string>{
+  static async updateCourseChannels(guildID: string, courseChannels: CourseChannels): Promise<string> {
     const config = await this.getForId(guildID);
     config.courseChannels = courseChannels;
     return this.update(config);
   }
+
+  static async create(data: { guildID: string, roles: Record<string, string> }): Promise<string> {
+    console.log('create call');
```
Comment (DrunkenToast) [inline src/services/guild-service.ts]: Remove unused logs

File: src/services/guild-service.ts [hunk 2]
```diff
@@ -25,9 +25,26 @@ export class GuildService {
...
   }
 
-  static async updateCourseChannels(guildID: string, courseChannels: CourseChannels): Promise<string>{
+  static async updateCourseChannels(guildID: string, courseChannels: CourseChannels): Promise<string> {
     const config = await this.getForId(guildID);
     config.courseChannels = courseChannels;
     return this.update(config);
   }
+
+  static async create(data: { guildID: string, roles: Record<string, string> }): Promise<string> {
```
Comment (DrunkenToast) [inline src/services/guild-service.ts]: More descriptive function name perhaps.
Maybe 'setDefault' or something along those lines.

File: src/discord.ts [hunk 3]
```diff
@@ -117,30 +118,41 @@ export async function buildClient(shard: number, shardCount: number): Promise<Cl
...
+    const roleIDs: Record<string, string> = {};
+    for (const roleName of roleNames) {
+      const role = await guild.roles.create({ data: { name: roleName } });
+      roleIDs[roleName] = role.id;
+    }
+    GuildService.create({ guildID: guild.id, roles: roleIDs }).catch(() => console.error('error'));
+
+    return;
+
+
```
Comment (DrunkenToast) [inline src/discord.ts]: \n\n\n\n\n\n

File: src/discord.ts [hunk 4]
```diff
@@ -117,30 +118,41 @@ export async function buildClient(shard: number, shardCount: number): Promise<Cl
...
       // eslint-disable-next-line no-await-in-loop
       const response = typeof command.response === 'function' ? await command.response(msg, guildConfig) : command.response;
 
       if (typeof response === 'string') {
-        msg.channel.send(response);
+        msg.channel.send(response).catch((err) => console.log(err));
         return;
       } else if (typeof response !== 'undefined') {
-        msg.channel.send(new MessageEmbed(response));
+        msg.channel.send(new MessageEmbed(response)).catch(err => console.error(err));
```
Comment (DrunkenToast) [inline src/discord.ts]: Properly log or throw

**Review by DrunkenToast — APPROVED** [review]

**Review by xunafay — APPROVED** [review]

**Review by xunafay — APPROVED** [review]

**Pull request merged**